### ISSUE TITLE
Replace all deprecated `sets.String` with `sets.Set`

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes/kubernetes#113709]
In this PR https://github.com/kubernetes/kubernetes/pull/112377, `sets.Set` is newly implemented, and `sets.String` got marked as deprecated.

In the scheduler, we often use `sets.String` everywhere. This issue aims to replace all `sets.String` (+ other deprecated structs in the `sets` package) with `sets.Set` in the sig/scheduling related area.

You can refer to the following comments.
https://github.com/kubernetes/kubernetes/blob/7158e0a4ffffcf50ce7e2f6de0c25fbb414b62b0/staging/src/k8s.io/apimachinery/pkg/util/sets/string.go#L23-L24

/kind cleanup
/sig scheduling
/good-first-issue

The sig/scheduling related area (I'm not sure if all of them use `sets.String` in it.):

 - [ ] `pkg/scheduler` 
 - [ ] `test/e2e/scheduling`
 - [ ] `test/integration/scheduler`
 - [ ] `test/integration/scheduler_perf`
 - [ ] `cmd/kube-scheduler`

**Comment by BenTheElder** [comment]
Please consider if these codepaths are likely to need cherry-picking to past releases, as only 1.25+ supports generics (this is noted in the generics policy linked in the previous PR)

**Comment by Aut0R3V** [comment]
I believe that all the instances only in that file need to be changed @sanposhiho? Thanks in advance!

**Comment by sanposhiho** [comment]
I may not grasp what you mean, but all the deprecated structs (+ their methods) defined in the `sets` packages should be replaced with new `sets.Set`. And, the target files are all files under pkg/scheduler, test/e2e/scheduling,  test/integration/scheduler, test/integration/scheduler_perf, and cmd/kube-scheduler. (But, as @BenTheElder, we may need to keep the deprecated struct in some code.)

**Comment by Aut0R3V** [comment]
Yea this just cleared up my doubt. Apologies for the ambiguous question.

**Comment by Bryce-Soghigian** [comment]
@tangwz yes I am still working on it

**Comment by Thenitesh143** [comment]
Since this project is not merged yet  so I am taking the opportunity to work on it.
Thank you

**Comment by BenTheElder** [comment]
The guidance on generics has been updated to target once 1.24 is the oldest supported version which will be true in January approximately (sometime after 1.26 releases and 1.23 heads out of support). I'd wait until then at this point.

**Comment by Bryce-Soghigian** [comment]
OK i can leave the PR open till january and rebase it then, when generics have full support.

**Comment by sanposhiho** [comment]
@BenTheElder Thanks for the information. Then, should we remove the deprecation notice on each set struct for now, or add some commend about the generics guidance on `sets.Set` for the Kubernetes internal contributors.

TBH, I didn't know about the [documentation](https://github.com/kubernetes/community/blob/master/sig-architecture/generics.md) about the usage of generics until you let us know. If I still didn't know that and got received the review request for a PR that included `sets.Set`, then I would put `/lgtm` on it after the regular reviewing.

**Comment by Bryce-Soghigian** [comment]
hey @glhl86  the work for this Issue has been done but as mentioned in the discussion above is awaiting generics support in all supported versions of k8s. 

If you are looking for projects to contribute to in k8s you can check out https://github.com/Bryce-Soghigian/good-first-issues-k8s Which i use to find issues tagged with good first issue or help wanted across all of the Kubernetes organization.

**Comment by Bryce-Soghigian** [comment]
Seeing as we just bumped everything up to 1.18 all supported versions right now should have generics. Can we rebase and merge?

**Comment by Ash-2k3** [comment]
@Bryce-Soghigian I am a novice and want to get started with contributing to open source. Are you still working on this issue ?

**Comment by Bryce-Soghigian** [comment]
@DeeptanshuDas  I have already done the work just waiting on 1.23 to become a legacy version.

**Comment by miank** [comment]
@DeeptanshuDas Have you finished working on this issue ?

**Comment by ashutosh887** [comment]
let me work on this issue now please
/assign

**Comment by abhishekkhanna2710** [comment]
Can I work on this issue now please
/assign

**Comment by DeeptanshuDas** [comment]
> @DeeptanshuDas Have you finished working on this issue ?

its showing something error  you can try this

**Comment by kidddddddddddddddddddddd** [comment]
@abhishekkhanna2710 split your commit by sig for better review and approval, and create a pr with that. 

> I have changed this PLease check @k8s-ci-robot

**Comment by abhishekkhanna2710** [comment]
> @abhishekkhanna2710 split your commit by sig for better review and approval, and create a pr with that.
> 
> > I have changed this PLease check @k8s-ci-robot

Can you tell me what it means I am new on this

**Comment by kidddddddddddddddddddddd** [comment]
> Can you tell me what it means I am new on this

Like Aldo said here: https://github.com/kubernetes/kubernetes/pull/116777#issuecomment-1477755352, split that pr by different sigs so a single pr will involve less reviewers.

**Comment by sarab97** [comment]
Is anyone working on this or is it available to take up

**Comment by guohaonan-shy** [comment]
let me work on this issue now please
/assign

**Comment by guohaonan-shy** [comment]
> @guohaonan-shy my description about my PR has been rejected, so I closed and rewrote. I have already finished my coding.

**Comment by guohaonan-shy** [comment]
@sarab97  OK,  I had some issues about my PR's description and therefore closed.

**Comment by Rohitkumar0107** [comment]
hey, I am looking for my first contribution in open source It seems easy Can i work on it.

**Comment by sarab97** [comment]
This issue is completely closed in PR #116940. The PR is already approved. It will be merged once the code freeze is lifted. No further work in regards to this issue is required.
